apps/backend/src/libs/hooks/check-user-permissions.hook.ts

```diff
@@ -1,13 +1,20 @@
-import { type FastifyRequest } from "fastify";
-
 import { ExceptionMessage } from "~/libs/enums/enums.js";
 import { checkHasPermission } from "~/libs/helpers/helpers.js";
-import { type APIPreHandler } from "~/libs/modules/controller/controller.js";
+import {
+	type APIHandlerOptions,
+	type APIPreHandler,
+} from "~/libs/modules/controller/controller.js";
 import { HTTPCode, HTTPError } from "~/libs/modules/http/http.js";
+import { type UserAuthResponseDto } from "~/modules/users/users.js";
 
-const checkUserPermissions = (routePermissions: string[]): APIPreHandler => {
-	return (request: FastifyRequest, _, done): void => {
-		const { user } = request;
+const checkUserPermissions = (
+	routePermissions: string[],
```

Comment:
let's pass it this way, also let's pass object instead of separate parameters here
```suggestion
	permissions: PermissionKey[],
	projectPermissions: ProjectPermissionKey[]
```

Reply:
So should I separate Permission Keys? Like now on Permission Key we have all permissions. Should I separate it everywhere or just for this one hook?

Reply:
now i just did 
```
const checkUserPermissions = (
	permissions: ValueOf<typeof PermissionKey>[],
	getProjectId?: (options: APIHandlerOptions) => number | undefined,
```

Reply:
yes, I mean projectPermissions and permissions should be passed separately and inside the checkPermission hook should be checked separately. So I mean I want it to be this way:
```
const hasProjectPermission = // check project permission
const hasUserPermission = // check user permission
const hasPermission =  hasProjectPermission || hasUserPermission;
```
And the same flow I want to see on frontend